Source/OxyPlot/Net40Compatibility/CompatibilityExtensions.cs

```diff
@@ -0,0 +1,55 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="CompatibilityExtensions.cs" company="OxyPlot">
+//   Copyright (c) 2020 OxyPlot contributors
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace OxyPlot
+{
+#if NET40
+    using OxyPlot.Net40Compatibility;
+#endif
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Provides extension methods facilitating compatibility with .NET 4.0.
+    /// </summary>
+    public static class CompatibilityExtensions
+    {
+        /// <summary>
+        /// Returns a <see cref="IReadOnlyList{T}"/> corresponding to the specified <see cref="List{T}"/>.
+        /// </summary>
+        /// <typeparam name="T">The type of the list.</typeparam>
+        /// <param name="list">The list.</param>
+        /// <returns>
+        /// On .NET4.0, this returns a wrapper around the <paramref name="list"/> which implements <see cref="IReadOnlyList{T}"/> and <see cref="IList{T}"/>.
+        /// On .NET4.5 and later, this directly returns the <paramref name="list"/>.
+        /// </returns>
+        public static IReadOnlyList<T> AsReadOnlyList<T>(this List<T> list)
```

Comment:
Is there an issue with just having one overload for `IList<T>`?

Reply:
Yes, as `IList<T>` does not implement `IReadOnlyList<T>`, we can't directly assign a `IList<T>` to a `IReadOnlyList<T>` (hopefully this madness will stop with .NET 5...)